api/pkg/s3/bucketlifecycledelete.go

```diff
@@ -20,49 +20,58 @@ import (
 
 	"github.com/emicklei/go-restful"
 	log "github.com/sirupsen/logrus"
-	"github.com/opensds/multi-cloud/api/pkg/common"
+	. "github.com/opensds/multi-cloud/s3/pkg/exception"
 	"github.com/opensds/multi-cloud/s3/proto"
+        "github.com/opensds/multi-cloud/api/pkg/common"
 )
 
 func (s *APIService) BucketLifecycleDelete(request *restful.Request, response *restful.Response) {
 	ctx := common.InitCtxWithAuthInfo(request)
+	bucketName := request.PathParameter("bucketName")
+	ruleID := request.Request.URL.Query()["ruleID"]
+	log.Infof("received request for deleting lifecycle rule[id=%s] for bucket[name=%s].\n", ruleID, bucketName)
+
+	if ruleID == nil {
+		response.WriteErrorString(http.StatusBadRequest, NoRuleIDForLifecycleDelete)
+		return
+	}
 
 	//var foundID int
 	FoundIDArray := []string{}
 	NonFoundIDArray := []string{}
-	bucketName := request.PathParameter("bucketName")
-	ruleID := request.Request.URL.Query()["ruleID"]
-	if ruleID != nil {
-		bucket, _ := s.s3Client.GetBucket(ctx, &s3.Bucket{Name: bucketName})
-		for _, id := range ruleID {
-			isfound := false
-			for _, lcRule := range bucket.LifecycleConfiguration {
-				if lcRule.Id == id {
-					isfound = true
-					FoundIDArray = append(FoundIDArray, id)
-				}
-			}
-			if !isfound {
-				NonFoundIDArray = append(NonFoundIDArray, id)
+
+	bucket, err := s.s3Client.GetBucket(ctx, &s3.BaseRequest{Id: bucketName})
+	if err != nil {
+		log.Infof("get bucket[name=%s] failed, err=%v.\n", bucketName, err)
+		response.WriteError(http.StatusInternalServerError, NoSuchBucket.Error())
```

Comment:
How about bucket exists but querying has failed, maybe because of network failure..in this case the NoSuchBucket Error maybe misleading

Reply:
Thanks for your suggestion, and shufang will open an issue to track the problem of bucket lifecycle delete here. the problem will be fixed in that issue.

Reply:
Here is the issue: https://github.com/opensds/multi-cloud/issues/192